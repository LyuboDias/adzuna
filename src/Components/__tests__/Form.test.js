/**
 * @jest-environment jsdom
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect"; // adding jest DOM
// additional functions we are going to use for our testing
import { render, screen, cleanup } from "@testing-library/react";
// importing the component that we going to run test on
// for testing we run in terminal 'npm run test'
import ContactForm from "../Form";
// using renderer as snapshot to verify that the component hasnt changed since last test
import renderer from "react-test-renderer";
// import { act, renderHook } from "@testing-library/react-hooks";

// clean up tree after every test
afterEach(() => {
  cleanup();
});

// testing if component renders properly
test("should render Form component", () => {
  render(<ContactForm />); // rendering the component
  const formElement = screen.getByTestId("contact-form"); // get component by id from the tree
  expect(formElement).toBeInTheDocument();
});

// renderer test with snapshot
test("matches snapshot", () => {
  const tree = renderer.create(<ContactForm />).toJSON();
  expect(tree).toMatchSnapshot(); // creating snapshot of the entire component
  // when component change and test fail if we happy with changes can update the snapshot with 'npm test -- -u' command
});
