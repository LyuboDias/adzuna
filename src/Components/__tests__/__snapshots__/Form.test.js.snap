// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sm:flex content-center "
  data-testid="contact-form"
>
  <div
    className="my-auto h-content bg-gray-100 rounded  max-h-sm py-12 px-10  max-w-sm"
  >
    <h3
      className="MuiTypography-root text-center text-blue-900  pb-6 font-light MuiTypography-h3"
    >
      Contact Form
    </h3>
    <form
      autoComplete="off"
      onSubmit={[Function]}
      title="contact form"
    >
      <label
        htmlFor="fullName"
        id="fullName"
      >
        Full Name
      </label>
      <input
        aria-describedby="enter full name"
        className="w-full border-2 py-4 border-blue-900 bg-gray-100 px-4 rounded my-3 focus:outline-none"
        name="fullName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Linus Torvalds"
        title="enter full name"
        type="text"
      />
      <span
        className="MuiTypography-root  text-red-500 font-light MuiTypography-caption"
      />
      <label
        htmlFor="email"
        id="email"
      >
        Email
      </label>
      <input
        aria-describedby="enter your email"
        className="w-full border-2 py-4 border-blue-900 bg-gray-100 px-4 rounded my-3 focus:outline-none"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="ltorvalds@linux.com"
        title="enter your email"
        type="text"
      />
      <span
        className="MuiTypography-root  text-red-500 font-light MuiTypography-caption"
      />
      <div
        className="relative"
      >
        <label
          htmlFor="description"
          id="description"
        >
          Your message
        </label>
        <textarea
          aria-describedby="enter your message here"
          className="w-full border-2 py-5 border-blue-900 bg-gray-100 px-4 rounded my-3 focus:outline-none "
          cols="50"
          id="description"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Linux is powerful, flexible, secure, reliable, stable, funâ€¦ than Windows."
          rows="4"
          title="enter your message here"
        />
        <span
          className="MuiTypography-root  text-red-500 font-light MuiTypography-caption"
        />
        <span
          className="text-gray-500 absolute -bottom-2 right-0"
          id="counter"
        />
      </div>
      <br />
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedSecondary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="submit button"
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root text-center text-white font-light MuiTypography-button"
          >
            Submit
          </span>
        </span>
      </button>
    </form>
  </div>
  <div
    className="hidden"
    style={
      Object {
        "opacity": 0,
        "transform": "scale(0.75, 0.5625)",
        "transformOrigin": "0 0 0",
        "visibility": "hidden",
      }
    }
  >
    <div
      className=" h-content bg-gray-100 rounded  max-h-sm py-4 px-10  max-w-sm "
    >
      <h4
        className="MuiTypography-root text-center text-3xl text-blue-900  mb-4 font-light MuiTypography-h4"
      >
        Thank You
      </h4>
      <h6
        className="MuiTypography-root text-center text-blue-900 mb-2 font-light MuiTypography-subtitle1"
      >
        We have recived your from
      </h6>
    </div>
  </div>
</div>
`;
